src/modules/sponsor/setEdition.ts

```diff
@@ -22,16 +23,29 @@ const promptSwitchToAccount = async (account: string, initial: boolean) => {
 export default async (edition: string) => {
   const previousConf = conf.getAll()
   const previousAccount = previousConf.account
+  const previousWorkspace = previousConf.workspace
+
   const sponsorClient = new Sponsor(getIOContext(), IOClientOptions)
   const data = await sponsorClient.getSponsorAccount()
   const sponsorAccount = R.prop('sponsorAccount', data)
+
   if (!sponsorAccount) {
+    if (previousWorkspace !== 'master') {
+      throw new CommandError('Can only set initial edition in master workspace')
+    }
     await promptSwitchToAccount('vtex', true)
-  } else if (previousAccount !== sponsorAccount) {
+  } else {
```

Comment:
Should we still `promptSwitchToAccount` when `previousAccount == sponsorAccount`?

Reply:
This is not a possible scenario, an account cannot be sponsor of itself. The previous check was quite misleading IMO so I decided to remove it.

If you think it's necessary though, I could check if they are the same to explode explicitly here, but I feel this 💥 should be somewhere else, not here (and it's probably already done).